online_meetings/bluejeans_report.py

```diff
@@ -0,0 +1,186 @@
+    
+    url = "https://indigo-api.bluejeans.com/v1/enterprise/" + str(ENV.get('BLUEJEANS_ENTERPRISE_ID')) + "/indigo/reportJob/meetings?filter=\
+    [{\"type\":\"date\",\"comparison\":\"lt\",\"value\":\"" + end.astimezone().isoformat() + "\",\"field\":\"end_time\"},\
+    {\"type\":\"date\",\"comparison\":\"lt\",\"value\":\"" + end.astimezone().isoformat() + "\",\"field\":\"start_time\"},\
+    {\"type\":\"date\",\"comparison\":\"gt\",\"value\":\"" + start.astimezone().isoformat() + "\",\"field\":\"start_time\"}]\
+    &fileName=meetings_30th-Mar-2020_30th-Mar-2020&userid=" + str(ENV.get("BLUEJEANS_USER_ID")) + "&app_name=command_center"
+
+    # post the report request job
+    payload = {}
+    headers = {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json',
+        'Authorization': f'Bearer {token}'
+    }
+    reportJobs_response = requests.request("POST", url, headers=headers, data=payload)
+    reportJob_json = json.loads(reportJobs_response.text.encode('utf8'))
+    if ("status" in reportJob_json) and (reportJob_json["status"] == "success") and ("jobId" in reportJob_json):
+        # get the job id
+        job_id = reportJob_json["jobId"]
+        logger.info("report job id = " + job_id)
+
+    # sleep to wait for report job finish
+    download_file_path = ""
+    while not download_file_path :
+        # retrieve report job information including the file download path
+        jobs_url = f"https://indigo-api.bluejeans.com/v1/enterprise/{ENV.get('BLUEJEANS_ENTERPRISE_ID')}/indigo/jobs/list?userid={ENV.get('BLUEJEANS_USER_ID')}"
+        payload = {}
+        headers = {
+            'Accept': 'application/json',
+            'Content-Type': 'application/json',
+            'Authorization': f'Bearer {token}'
+        }
+        jobs_response = requests.request("GET", jobs_url, headers=headers, data=payload)
+        jobs_json = json.loads(jobs_response.text.encode('utf8'))
+        for job_json in jobs_json:
+            # find the right job:
+            if job_json['_id'] == job_id:
+                if (job_json['status'] == 'success'):
+                    download_file_path = job_json['message']['path']
+                    logger.info(f"job {job_id} finished yet: {job_json['status']} and download file path is {download_file_path}")
+                    break
+                else:
+                    logger.info(f"job {job_id} not finished yet: {job_json['status']}")
+                    time.sleep(10)
+
+    download_url = f"https://indigo-api.bluejeans.com/v1/enterprise/{ENV.get('BLUEJEANS_ENTERPRISE_ID')}/indigo/{download_file_path}"
+
+    # download the report file
+    payload = {}
+    headers = {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json',
+        'Authorization': f'Bearer {token}'
+    }
+    r = requests.get(download_url, headers=headers)
+
+    # specifying the zip file name
+    zip_file_name = "./bj.zip"
```

Comment:
This won't work in the container. We're probably going to have to create a storage volume, pass in a path via ENV and use that to write. (Like /tmp for instance)

Reply:
New issue as #55